.github/workflows/release-rc.yml

```diff
@@ -1,18 +1,48 @@
 name: Release Armada components - RC
 
 on:
-  push:
+  workflow_run:
+    types: [completed]
+    workflows: [CI]
     branches:
-      - main
       - master
 
 permissions:
   contents: write
 
 jobs:
+  validate:
+    if: github.event.workflow_run.event == 'push' && github.event.workflow_run.conclusion == 'success' && github.repository_owner == 'armadaproject'
+    name: "Validate revision"
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v3"
+        with:
+          fetch-depth: 0
+
+      - name: Validate ref
```

Comment:
You should add a comment explaining the logic here, e.g.
```
      # The given ref should belong to the master branch and it shouldn't
      # be more than 2 commits away (in case another push happened in the meantime).
      # Anything else is invalid.
```